### PR TITLE
Migrate setWeek, setWeekYear functions

### PR DESCRIPTION
**Pull request by maximtop**
This pr 
fixes: https://github.com/date-fns/date-fns-typescriptify/issues/133 - setWeek
fixes: https://github.com/date-fns/date-fns-typescriptify/issues/134 - setWeekYear

### CODE REVIEW

**Review by kossnocorp — CHANGES_REQUESTED** [review]
Looks good, but I think it worth extracting type alias.

File: src/types.ts [hunk 1]
```diff
@@ -23,6 +23,10 @@ export interface WeekStartOptions {
   weekStartsOn?: Day
 }
 
+export interface WeekYearStartOptions {
+  firstWeekContainsDate?: 1 | 2 | 3 | 4 | 5 | 6 | 7
+}
+
```
[the remark this answers is not in the document]
Comment (maximtop) [reply src/types.ts]: @kossnocorp i've extracted

File: src/setWeekYear/index.ts [hunk 2]
```diff
@@ -3,6 +3,7 @@ import startOfWeekYear from '../startOfWeekYear/index'
 import toDate from '../toDate/index'
 import toInteger from '../_lib/toInteger/index'
 import requiredArgs from '../_lib/requiredArgs/index'
+import { LocaleOptions, WeekStartOptions, WeekYearStartOptions } from '../types';
```
Comment (kossnocorp) [inline src/setWeekYear/index.ts]:
```suggestion
import { LocaleOptions, WeekStartOptions, WeekYearStartOptions } from '../types'
```

File: src/setWeekYear/index.ts [hunk 3]
```diff
@@ -49,26 +50,30 @@ import requiredArgs from '../_lib/requiredArgs/index'
  * })
  * //=> Sat Jan 01 2005 00:00:00
  */
-export default function setWeekYear(dirtyDate, dirtyWeekYear, dirtyOptions) {
+export default function setWeekYear(
+    dirtyDate: Date | number,
+    dirtyWeekYear: number,
+    dirtyOptions?: LocaleOptions & WeekStartOptions & WeekYearStartOptions,
```
Comment (kossnocorp) [inline src/setWeekYear/index.ts]:
```suggestion
    options: LocaleOptions & WeekStartOptions & WeekYearStartOptions = {},
```

File: src/setWeekYear/index.ts [hunk 4]
```diff
@@ -49,26 +50,30 @@ import requiredArgs from '../_lib/requiredArgs/index'
...
+): Date {
   requiredArgs(2, arguments)
 
-  var options = dirtyOptions || {}
-  var locale = options.locale
-  var localeFirstWeekContainsDate =
+  const options = dirtyOptions || {}
```
Comment (kossnocorp) [inline src/setWeekYear/index.ts:60]:
```suggestion
```

**Review by kossnocorp — APPROVED** [review]

**Comment by kossnocorp** [comment]
Something funny happened to GitHub. Dismiss my review up to this point.

File: src/types.ts [hunk 5]
```diff
@@ -23,6 +23,12 @@ export interface WeekStartOptions {
   weekStartsOn?: Day
 }
 
+export type firstWeekContainsDate = 1 | 2 | 3 | 4 | 5 | 6 | 7
+
+export interface WeekYearStartOptions {
+  firstWeekContainsDate?: firstWeekContainsDate
+}
+
```
Comment (kossnocorp) [inline src/types.ts:31]: Please sync with @tan75 https://github.com/date-fns/date-fns/pull/2375/files#r603154585

**Review by kossnocorp — CHANGES_REQUESTED** [review]

**Review by kossnocorp — APPROVED** [review]
✌

**Pull request merged**